### PR TITLE
Fe edit  profile info

### PR DESCRIPTION
**Pull request by annaBalabak**

### CODE REVIEW

File: frontend/src/services/userData/editUserData.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import axios from 'axios';
+import { updateUser } from '../../redux/slices/userSlice';
+
+export const editUserData = async (updatedData, dispatch) => {
+  const id = localStorage.getItem('authId');
+  const token = localStorage.getItem('authToken');
+  try {
+    const response = await axios.put(
+      `https://api.imperiaholoda.com.ua:4446/v1/users/${id}`,
```
Comment (Svitlana-G) [inline frontend/src/services/userData/editUserData.js:9]: You can use axiosInstance:` const response = await axiosInstance.put(`/users/${id}`, updatedData)`
То make everything work, you need to replaсe token -> authToken   on line 17 in httpClient.js

File: frontend/src/components/account/ChangeDetails/ChangeDetails.jsx [hunk 2]
```diff
@@ -1,10 +1,150 @@
...
+    setCorrectUserData({
+      firstName: userData.firstName || '',
+      lastName: userData.lastName || '',
+      surname: userData.surname || '',
+      email: userData.email || '',
+      phone: userData.phone || '',
+      birthday: userData.birthday || '',
+    });
+  }, [userData]);
+
```
[the remark this answers is not in the document]
Comment (annaBalabak) [reply frontend/src/components/account/ChangeDetails/ChangeDetails.jsx:34]: in useState I initialize the state with the initial values from userData and  useEffect is used to update the state whenever userData changes

File: frontend/src/components/login/ResetPassword/ResetPasswordPage.module.scss [hunk 3]
```diff
@@ -63,7 +63,7 @@
   span {
     color: $discount-color;
   }
-}
+} */
```
[the remark this answers is not in the document]
Comment (annaBalabak) [reply frontend/src/components/login/ResetPassword/ResetPasswordPage.module.scss:66]: done

File: frontend/src/components/account/ChangePassword/ChangePassword.jsx [hunk 4]
```diff
@@ -1,11 +1,51 @@
...
+              />
+            </div>
+
+            <div className={styles.btnContainer}>
+              <Button
+                variant="primary"
+                label="Підтвердити"
+                padding="padding-sm"
+                type="submit"
+                /* onClick={handleSubmit} */
```
[the remark this answers is not in the document]
Comment (annaBalabak) [reply frontend/src/components/account/ChangePassword/ChangePassword.jsx:36]: removed

[the remark this answers is not in the document]
Comment (annaBalabak) [reply frontend/src/components/account/ChangePassword/ChangePassword.jsx:36]: removed

File: frontend/src/services/userData/editUserData.js (hunk 1 above)
Comment (annaBalabak) [reply frontend/src/services/userData/editUserData.js:9]: done

File: frontend/src/components/account/ChangeDetails/ChangeDetails.jsx [hunk 5]
```diff
@@ -1,10 +1,150 @@
...
+          <div className={styles.formContainer}>
+            <div className={styles.nameData}>
+              <div className={styles.dataContainer}>
+                <label htmlFor="firstName">Ім’я</label>
+                <input
+                  id="firstName"
+                  className={styles.dataInput}
+                  type="text"
+                  value={correctUserData.firstName}
+                  onChange={handleChange}
```
Comment (halynarydasova) [inline frontend/src/components/account/ChangeDetails/ChangeDetails.jsx:71]: Please add placeholders for each field

File: frontend/src/components/ui-kit/components/PasswordForm/PasswordForm.module.scss [hunk 6]
```diff
@@ -0,0 +1,48 @@
+@import '../../../../variables';
+@import '../../../../mixins';
+
+/* .formContainer {
+  display: flex;
```
Comment (halynarydasova) [inline frontend/src/components/ui-kit/components/PasswordForm/PasswordForm.module.scss:5]: Please delete unnecessary styles

File: frontend/src/components/account/ChangeDetails/ChangeDetails.jsx (hunk 5 above)
Comment (annaBalabak) [reply frontend/src/components/account/ChangeDetails/ChangeDetails.jsx:54]: done

File: frontend/src/components/account/ChangePassword/ChangePasswordForm.jsx [hunk 7]
```diff
@@ -1,42 +1,143 @@
...
       <Account title="Редагування особистої інформації">
         <div className={styles.formContainer}>
           <h2 className={styles.title}> Зміна паролю</h2>
-          <form action="">
+          <form onSubmit={handlePasswordChange}>
+            <div className={styles.passwordContainer}>
+              <label htmlFor="password">Введіть старий пароль</label>
+              <input
+                className={styles.dataInput}
+                id="oldPassword"
```
Comment (halynarydasova) [inline frontend/src/components/account/ChangePassword/ChangePasswordForm.jsx:67]: Do you think there's a need to use controlled input here? As I understand it, you only need this info on submission. Maybe it's better to use ref here

File: frontend/src/components/account/ChangePassword/ChangePasswordForm.module.scss [hunk 8]
```diff
@@ -28,3 +28,15 @@
   gap: 32px;
   margin-top: 32px;
 }
+
+.passwordReq {
+  color: $gray-dark-color;
```
Comment (halynarydasova) [inline frontend/src/components/account/ChangePassword/ChangePasswordForm.module.scss:33]: Please, add some space between this hint and the input

Comment (annaBalabak) [reply frontend/src/components/account/ChangePassword/ChangePasswordForm.module.scss:33]: There is no space according to the design (you can check design at registration modal)

File: frontend/src/components/account/ChangePassword/ChangePasswordForm.jsx (hunk 7 above)
Comment (annaBalabak) [reply frontend/src/components/account/ChangePassword/ChangePasswordForm.jsx:69]: done

**Review by halynarydasova — APPROVED** [review]

**Pull request merged**